lib/puppet/pops/types/p_object_type.rb

```diff
@@ -333,6 +372,86 @@ def initialize(name_or_i12n_hash, i12n_hash_expression = nil)
     end
   end
 
+  # @api private
+  def new_function(loader)
+    @new_function ||= create_new_function(loader)
+  end
+
+  # @api private
+  def create_new_function(loader)
```

Comment:
Yes, I know (I just said so). Something needs to be said about this in the `new` function until we have a better way.